### PR TITLE
[jit] Add binary cross entropy to standard lib

### PR DESCRIPTION
**Pull request by driazati**

### CODE REVIEW

File: torch/csrc/jit/register_special_ops.cpp [hunk 1]
```diff
@@ -83,6 +83,36 @@ RegisterOperators reg({
             push(stack, ss.str());
             return 0;
           };
+        }),
+    Operator(
+        "aten::nelement(Tensor self) -> int",
```
Comment (eellison) [inline torch/csrc/jit/register_special_ops.cpp]: Can't you use numel?

File: torch/jit/__init__.py [hunk 2]
```diff
@@ -1386,6 +1386,7 @@ def register_all(mod):
     _builtin_table[id(_list_with_default)] = "aten::list_with_default"
     _builtin_table[id(_unwrap_optional)] = "aten::_unwrap_optional"
     _builtin_table[id(cudnn.is_acceptable)] = "aten::cudnn_is_acceptable"
+    _builtin_table[id(torch._C._infer_size)] = "aten::_infer_size"
```
Comment (eellison) [inline torch/jit/__init__.py]: Could you just write at::infer_size and not add the extra operator ?

File: torch/csrc/jit/register_special_ops.cpp [hunk 3]
```diff
@@ -83,6 +83,36 @@ RegisterOperators reg({
...
+            int nelements = 1;
+            for (auto size : t.sizes()) {
+              nelements *= size;
+            }
+
+            push(stack, nelements);
+            return 0;
+          };
+        }),
+    Operator(
```
Comment (eellison) [inline torch/csrc/jit/register_special_ops.cpp]: Could you add extra tests for both ops in test/test_jit.py ?

File: torch/jit/__init__.py (hunk 2 above)
Comment (driazati) [reply torch/jit/__init__.py]: There's no `aten` op for `infer_size` (plus we have 2 overloads of `infer_size` in `ExpandUtils.cpp` and in `InferSize.h`), so I think we have to have an op for it

File: torch/csrc/jit/register_special_ops.cpp [hunk 4]
```diff
@@ -38,7 +39,6 @@ RegisterOperators reg({
     Operator(
         "aten::list_with_default(int[] list, int[] defaults) -> int[]",
         [](Stack& stack) {
-          autograd::profiler::RecordFunction record("sizes");
```
Comment (eellison) [inline torch/csrc/jit/register_special_ops.cpp]: Could you remove this so my PR & yours don't conflict ?

**Review by eellison — APPROVED** [review]
LGTM. In a future diff we can move _infer_size to aten

**Pull request closed without merging**